docs/assets/js/_partials/chart-area.js

```diff
@@ -0,0 +1,251 @@
+class AreaChart
+{
+    constructor(config, canvas, actionBar)
+    {
+        this.config = config;
+        this.actionBar = actionBar;
+
+        this.context = canvas.getContext('2d');
+        this.views = [];
+        this.defaultView = null;
+
+        // Show spinning animation while loading
+        this.spinner = createSpinner(canvas);
+    }
+
+    render(data)
+    {
+        sortTimeData(data);
+
+        this._checkData(data);
+        this._initializeViews(data);
+
+        // Build chart data for each view separately
+        for (const view of this.views)
+        {
+            this._aggregateData(view);
+
+            // Build the actual data used by Chart.js for rendering
+            this._buildChartData(view);
+        }
+
+        this.chartJSChart = new Chart(
+            this.context,
+            {
+                type: 'line',
+                data:
+                {
+                    datasets: this.defaultView.chartData,
+                },
+                options: this.defaultView.style,
+            });
+
+        this._renderActionBar();
+        this.spinner.stop();
+    }
+
+    _checkData(data)
+    {
+        for (const row of data)
+            if (!('date' in row))
+                throw 'invalid data format';
+    }
+
+    _initializeViews(data)
+    {
+        // Collect the data for each view in an array
+        if ('views' in this.config)
+        {
+            this.views = this.config.views;
+
+            for (const view of this.views)
+                view.rawData = deepCloneArray(data);
```

Comment:
In context of Git the function name `deepCloneArray` might be confusing. Maybe `deepCopyArray`?

Reply:
Well, shallow vs. deep cloning are standard programming terms I guess. I’d keep it like this, because I believe that everybody who’ll touch our source code will understand what’s meant here.

Reply:
See [my other comment below](https://github.com/Autodesk/hubble/pull/174#discussion_r238363234) :point_down:.